Lifter system

Goalaso opened the pull request:
Created methods for lifter to move up/down, also added a method that checks where the arm is (needs to be finished) and also check for when to start moving lifter up when arm is lifting a crate (to be finished). Also added a check for if the move to position calls for a position that is not within the bounds and then moves to upper/lower limit (depending on which way it's moving)

skallaher committed:
More updates to the config system with implementation of GamePad and
Cyborg channels.

skallaher committed:
Added necessary requires() to commands and set up Wheels.

skallaher committed:
Added to Arm and Lifter, forgot to set mode and enable control in
Wheels.

skallaher committed:
Major command update

skallaher committed:
Missed files for major command update. Also reworked config and added to
subsystems. Documentation.

skallaher committed:
Added utility methods to Wheels.

skallaher committed:
More outlining and documentation work.

Goalaso committed:
Added some lift commands and arm commmands (todo)

skallaher commented:
I recommend creating constants for any unknown values. Try to keep from using what are called magic numbers throughout code by putting them somewhere that they can easily be changed. (i.e. Arm.getInstance().moveArm(1); take the 1 and make it a final variable in the class.) Also remember to document everything. Finally, any sort of value that will be constant, such as the arm position to start moving the lifter at, should be made a final variable.

Goalaso committed:
Removed "magic numbers" and added final variables

Goalaso committed:
Added teleop arm command
Added getButton for arm up/down
TODO: figure out what button to use for arm up/down

skallaher commented:
Looks pretty good, I'll make some edits and it should be ready for testing.

skallaher committed:
Moved some magic numbers to final variables and reconfigured some
variable names. Also updated the arm to be position controlled.

skallaher commented:
Just noticed that this is intended to merge into master. Resubmit a pull request to merge with Dev.
